zigpy/zcl/foundation.py

```diff
@@ -322,9 +322,9 @@ def deserialize(cls, data):
         else:
             # Notifying that I will report things to you
             self.datatype, data = t.uint8_t.deserialize(data)
+            datatype = DATA_TYPES[self.datatype]
```

Comment:
Why moving the datatype away from the line were it is being used?

Reply:
I moved that line up during debug close to where self.datatype is assigned and had an instruction in place to log its value.
I initially examined the possibility that something was wrong with the datatype and as bytes are missing at some point, that exception occurred before I could look at datatype.

Do you know why datatype is named very much like self.datatype while they are not the same variable?  
Personnally I'ld avoid that.  It could be datatypePropse, typeProps, typeInfo, ... .